File: 100 days of leetcode/Day10.py
class Solution(object):
    def canPlaceFlowers(self, flowerbed, n):
        """
        :type flowerbed: List[int]
        :type n: int
        :rtype: bool
        """
        x = len(flowerbed)
        if(n == 0):
            return True
        
        for i in range(x):
            if n > 0:
                if (flowerbed[i] == 0) and (i == 0 or flowerbed[i-1] == 0 ) and (i == x - 1 or flowerbed[i+1] == 0):
                    flowerbed[i] = 1
                    i += 1
                    n -= 1
        
        if n == 0:
            return True
        return False
        

File: 100 days of leetcode/Day10.java
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        int x = flowerbed.length;
        // int n = x;

        // So first I will find empty plot and then I will check if the adjacent plots are also empty or not.
        // If yes then I will put a flower there and n--
        // Else I will go to next plot
        if(n == 0){
            return true;
        }

        for(int i = 0; i < x && n > 0; i++){
            if(flowerbed[i] == 0 && (i == 0 || flowerbed[i-1] == 0) && ( i == x -1 || flowerbed[i+1] == 0)){ 
                flowerbed[i] = 1;
                i++;
                n--;
            }
        }
        if(n == 0){
            return true;
        }
        else{
            return false;
        }
    }
}